### PR TITLE
docs: detect RC versions as preview

### LINKED ISSUE #204: Create the 'preview' shortcut next to 'current'
Have a fixed URL for the preview version next to 'current', which is not indexed by search engines.

### PR DESCRIPTION
**Pull request by gbartolini**
Add the '/preview/' directory for the latest RC version.

Add the version of CNPG on left sidebar.

Closes #204

### CODE REVIEW

File: hack/import_docs.sh [hunk 1]
```diff
@@ -27,14 +30,18 @@ BRANCH_NAME=release-$release_version
 # Use the main branch if we are not targeting a stable version
 if ! [[ "${release_version}" =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
     BRANCH_NAME="main"
+    PREVIEW_RELEASE=1
```
Comment (jsilvela) [inline hack/import_docs.sh:33]: So, for the release candidate we're getting the documentation from `main`.
Two concerns:
- This script should not be re-run, say, a few days after, as the docs on main may have changed
- There are some placeholders in the documentation generation where we show the branch name. In this case we'll see `main`, not `1.24.0-rc1`. These mentions are not super-prominent, but they're there

Comment (jsilvela) [reply hack/import_docs.sh:33]: Ah, no, you already thought of that, I think....

**Comment by jsilvela** [comment]
As far as I see this makes sense. Still need to test the script out.
Just the concern that the script will need to be applied essentially once since it pulls from main.
I guess in "RC mode" not a big concern, as we'll be mostly in code freeze.

**Comment by leonardoce** [comment]
I tested it manually. Sounds like everything is fine

**Comment by jsilvela** [comment]
Tested and even the version tag works fine.

Screenshot 2024-07-26 at 12 16 42

**Review by jsilvela — DISMISSED** [review]

File: hack/import_docs.sh [hunk 2]
```diff
@@ -44,22 +51,35 @@ popd
...
-current_version=$(ls $DOCDIR | sort -n | tail -1)
-if [ -d $DOCDIR/current ]
+# Preview release
+if [ $PREVIEW_RELEASE -eq 1 ]
 then
-    git rm -fr $DOCDIR/current
+  cat > $TARGETDIR/robots.txt <<EOF
+User-agent: *
+Disallow: /
+EOF
```
Comment (mnencia) [inline hack/import_docs.sh:64]: The created file will be visible as `/documentation/1.24.0rc1/robots.txt`.
AFAIR it's not supported to have a `robot.txt` file in a directory different from the root.

**Review by mnencia — APPROVED** [review]

File: assets/robots.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+User-agent: *
```
Comment (jsilvela) [inline assets/robots.txt:1]: Always create the robots.txt file?
If it's not going to have anything, better not to have it, no?

Comment (jsilvela) [reply assets/robots.txt:1]: IMO you could put this in the same place where the `Disallow` gets added. But, optional.

**Review by jsilvela — APPROVED** [review]

File: assets/robots.txt (hunk 3 above)
Comment (mnencia) [reply assets/robots.txt]: Done.

**Review by mnencia — APPROVED** [review]

**Review by leonardoce — APPROVED** [review]

**Comment by gbartolini** [comment]
Good catch on robots.txt @mnencia . fixation error here.

**Pull request merged**